dex/networks/eth/contractor.go

```diff
@@ -0,0 +1,192 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+//go:build lgpl
+// +build lgpl
+
+package eth
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"math/big"
+	"strings"
+	"time"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/encode"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// Contractor is a translation layer between the abigen bindings and the DEX app.
+// The intention is that if a new contract is implemented, the Contractor
+// interface itself will not require any updates.
+type Contractor interface {
```

Comment:
In general use for Contractor server-side?  why not in client/asset/eth?

Reply:
Good question. The only abigen method we're using server-side is `Swap`, and the server will only support one version at a time. I guess I should move it to **client/asset/eth** then.